ingestion/src/metadata/ingestion/models/custom_pydantic.py

```diff
@@ -30,13 +32,187 @@
 SECRET = "secret:"
 JSON_ENCODERS = "json_encoders"
 
+RESTRICTED_KEYWORDS = ["::", ">"]
+RESERVED_COLON_KEYWORD = "__reserved__colon__"
+RESERVED_ARROW_KEYWORD = "__reserved__arrow__"
+
+CREATE_ADJACENT_MODELS = ["ProfilerResponse", "SampleData"]
+
 
 class BaseModel(PydanticBaseModel):
     """
     Base model for OpenMetadata generated models.
     Specified as `--base-class BASE_CLASS` in the generator.
     """
 
+    @model_validator(mode="after")
+    @classmethod
+    def parse_name(  # pylint: disable=inconsistent-return-statements
+        cls, values
+    ):  # pylint: disable=too-many-locals
+        """
+        Primary entry point to process values based on their class.
+        Uses singledispatch to dynamically call appropriate processing functions.
+        """
+
+        from metadata.generated.schema.api.data.createDashboard import (
+            CreateDashboardRequest,
+        )
+        from metadata.generated.schema.api.data.createTable import CreateTableRequest
+        from metadata.generated.schema.api.data.createTableProfile import (
+            CreateTableProfileRequest,
+        )
+        from metadata.generated.schema.entity.data.table import (
+            Column,
+            Table,
+            TableConstraint,
+        )
+        from metadata.profiler.processor.core import ColumnProfile, ProfilerResponse
+        from metadata.profiler.source.metadata import ProfilerSourceAndEntity
+        from metadata.utils.entity_link import CustomColumnName
+
+        @singledispatch
+        def validate_name_and_transform(_):
+            """Single dispatch function to process name and display name based on object type."""
+
+        @validate_name_and_transform.register
+        def _(obj: CreateTableRequest):
+            """Process CreateTableRequest type."""
+            process_table_and_columns(obj, is_create=True)
+
+        @validate_name_and_transform.register
```

Comment:
Instead of implementing this on a per-entity basis might make more sense to hook into the serialization process and modify any `basic.EntityName`types.

When we add new entities in the future, one will need to explicitly add it here. Makes more sense to have this functionality baked in so that high-level implementation dont require any awareness to this logic.

Reply:
that's a good one actually. @ayush-shah similar to the comment above. Regardless of the object we can traverse its schema, and if it's instance of EntityName we just convert the ::